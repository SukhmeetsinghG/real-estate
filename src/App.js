import LogList from "./components/LogList";

export default function App() {

  return (
    <>
    <LogList />
    </>
  );
}
