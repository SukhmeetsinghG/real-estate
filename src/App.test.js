import { render, screen, fireEvent } from '@testing-library/react';
import App from './App';
import {BrowserRouter as Router, useNavigate} from 'react-router-dom';


test('renders learn react link', () => {
  render(<Router><App /></Router>);
  const linkElement = screen.getByText(/Log List/i);
  expect(linkElement).toBeInTheDocument();
});

test('renders logger system', () => {
  const {container} = render(<Router><App /></Router>);
  expect(container).toMatchSnapshot();
});

test("Test search button event", async () => {
  render(<Router><App /></Router>);
  const searchButton = screen.getByText(/Search/i);
  fireEvent.click(searchButton);
});