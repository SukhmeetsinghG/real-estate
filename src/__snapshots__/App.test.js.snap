// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders logger system 1`] = `
<div>
  <div
    class="py-5 paddingrl"
  >
    <h1>
      Log List
    </h1>
    <form
      action=""
    >
      <div
        class="d-flex"
      >
        <div
          class="form-group px-2"
        >
          <label>
            Log Id
          </label>
          <input
            class="form-control"
            name="logId"
            type="text"
            value=""
          />
        </div>
        <div
          class="form-group px-2"
        >
          <label>
            Application Type
          </label>
          <select
            class="form-control"
            name="applicationType"
          >
            <option
              value=""
            >
              Select Application Type
            </option>
          </select>
        </div>
        <div
          class="form-group px-2"
        >
          <label>
            Application Id
          </label>
          <input
            class="form-control"
            name="applicationId"
            type="text"
            value=""
          />
        </div>
        <div
          class="form-group px-2"
        >
          <label>
            Action Type
          </label>
          <select
            class="form-control"
            name="actionType"
          >
            <option
              value=""
            >
              Select Action Type
            </option>
          </select>
        </div>
        <div
          class="form-group px-2 "
        >
          <label>
            From Date
          </label>
          <div
            class="react-datepicker-wrapper"
          >
            <div
              class="react-datepicker__input-container"
            >
              <input
                class="form-control"
                name="from_date"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          class="px-2"
        >
          <label>
            To Date
          </label>
          <div
            class="react-datepicker-wrapper"
          >
            <div
              class="react-datepicker__input-container"
            >
              <input
                class="form-control"
                name="to_date"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          class="px-2 pt-4"
        >
          <button
            class="searchAction btn btn-primary"
            type="button"
          >
            Search
          </button>
        </div>
      </div>
    </form>
    <hr
      class="my-4"
    />
    <div
      class="card mt-2"
    >
      <div
        class="sc-dmctIk WQNxq table table-striped table-bordered"
      >
        <div
          class="sc-fLcnxK dApqnJ"
        >
          <div
            class="sc-bcXHqe kVrXuC rdt_Table"
            role="table"
          >
            <div
              class="sc-iveFHk bzRnkJ"
            >
              <div
                style="padding: 24px;"
              >
                There are no records to display
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
